modules/cli/test-res/selectionSort/selectionSort.dfy

```diff
@@ -22,7 +27,7 @@ method selectionSort(a: seq<int>) returns (r: seq<int>)
       invariant label yInBounds: x <= y <= |r|
       invariant label mInBounds: x <= m <= y && m < |r|
       invariant label mIsMin: forall l: int :: x <= l < y ==> r[m] <= r[l]
-//      invariant asBag(a) == asBag(r)
+      invariant multiset(a) == multiset(r)
     decreases |r| - y
     {
       if a[y] < a[m]
```

Comment:
Syntax error in line 40, 41. Test does not pass.

Reply:
Well, it seems to work in the (swing) gui
![Screenshot_2020-06-15_11-50-19](https://user-images.githubusercontent.com/17046629/84643838-a9af8500-aefe-11ea-836d-7b2c04e38eab.png)

Reply:
Well, I used "boogie;" as standard script. Perhaps that does the trick!

Reply:
What is the syntax error there? Can you paste the stacktrace?

Reply:
The problem is that sequences cannot be assigned like arrays but must be updated in the swap lines 39-41.
``// var t:int := r[m];``
``    // Following array like assignments not possible in sequences.``
``    // r[m] := r[x];``
``    // r[x] := t;``
 ``   // do instead``
``    r := r[m := r[x]][x:= r[m]];``

Reply:
It did not raise an error here. DId you get one? Stacktrace?

Reply:
I am sorry. The test passes.

In my opinion, it should not, though. I altered the file and the test would fail.
![dafnydoesnotpermit](https://user-images.githubusercontent.com/15802245/84674473-3ec87300-af2b-11ea-94dc-f69223f95089.png)